external-resolver/src/main/scala/org/genivi/sota/resolver/packages/Package.scala

```diff
@@ -20,44 +21,20 @@ case class PackageFilter(
   filterName    : Filter.Name
 )
 
-object Package {
-
-  case class Id(
-    name   : Package.Name,
-    version: Package.Version
-  )
+object Package extends PackageCommon {
 
   case class Metadata(
     description: Option[String],
     vendor     : Option[String]
   )
 
-  trait ValidName
-  trait ValidVersion
-  trait ValidId
-
-  type Name        = Refined[String, ValidName]
-  type Version     = Refined[String, ValidVersion]
-  type NameVersion = Refined[String, ValidId]
-
-  implicit val validPackageName: Predicate[ValidName, String] =
-    Predicate.instance( _.nonEmpty, _ => "Package name required" )
-
-  implicit val validPackageVersion: Predicate[ValidVersion, String] =
-    Predicate.instance( _.matches( """^\d+\.\d+\.\d+$""" ), _ => "Invalid version format")
-
-  implicit val validPackageId: Predicate[ValidId, String] =
-    Predicate.instance(s =>
-      {
-        val nv = s.split("-")
-        nv.length == 2 &&
-          validPackageName.isValid(nv.head) &&
-          validPackageVersion.isValid(nv.tail.head)
-      }
-      , s => s"Invalid package id (should be package name dash package version): $s")
-
-  implicit val PackageIdOrdering: Ordering[Package.Id] = new Ordering[Package.Id] {
-    override def compare(id1: Package.Id, id2: Package.Id): Int =
-      id1.name.get + id1.version.get compare id2.name.get + id2.version.get
-  }
+  val genPackage: Gen[Package] = for {
+    id      <- genPackageId
+    desc    <- Gen.option(Arbitrary.arbitrary[String])
+    vendor  <- Gen.option(Gen.alphaStr)
+  } yield Package(id, desc, vendor)
+
+  implicit val arbPackage: Arbitrary[Package] =
+    Arbitrary(genPackage)
```

Comment:
I think it makes sense keep instances in the companion objects (so they become available when the datatype is imported). 

Also generating a random package isn't really test code. Sure, that's what it will be used for in our case, but it could have other applications...